src/arrow/CMakeLists.txt

```diff
@@ -0,0 +1,66 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+if(NOT ICEBERG_ARROW)
```

Comment:
Why do we need a directory called arrow, ISTM we do need some helper functions to wrapper the usage of arrow, but put it directly under src make people think arrow is part of iceberg's spec. I'd suggest we create a directory 
called util and put arrow wrapper under that, what do you think?

Reply:
This is related to the discussion: https://github.com/apache/iceberg-cpp/issues/2#issuecomment-2522063550. In short, we intend to make the core library as lite as possible to depend on a minimal set of third-party libraries. Arrow is one of the choose to provide concrete I/O and Parquet support.